playwright: 1.34.3 > 1.37.0

pietdevries94 opened the pull request:
## Description of changes

Updated with the script in ./pkgs/development/python-modules/playwright/update.sh
Previous pr by bot only updated the main package, not the driver en web test package. https://github.com/NixOS/nixpkgs/pull/239517

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I have used it in an overlay and the update works.

natsukium reviewed (commented):
Thanks for your contribution!
The change looks good to me.

To prevent future inconsistencies, would you please add `passthru.updateScript = . /update.sh;` to the python module?

pietdevries94 commented:
> Thanks for your contribution! The change looks good to me.
> 
> To prevent future inconsistencies, would you please add `passthru.updateScript = . /update.sh;` to the python module?

I have added the line to the the other passthru stuff
Also, version 1.37.0 is out, so I updated it that step further.
The update script missed a jq import, which the script used, so I also added that

teto commented:
the passthru driver is bad, in one of our projects we need to override the playwright-test version but it doesnt impact the driver version (or vice-versa, dont remember). I've made this hackish change to avoid the silent mismatch but didn't have time to clean it up https://github.com/NixOS/nixpkgs/pull/249738

OliverJAsh commented:
Can we bump this to 1.37.1? https://github.com/microsoft/playwright/releases/tag/v1.37.1

pietdevries94 commented:
> the passthru driver is bad, in one of our projects we need to override the playwright-test version but it doesnt impact the driver version (or vice-versa, dont remember). I've made this hackish change to avoid the silent mismatch but didn't have time to clean it up #249738

Is there something I can do to help here?

> Can we bump this to 1.37.1? https://github.com/microsoft/playwright/releases/tag/v1.37.1

Because the nixpkgs version is coupled with playwright-python, that is not possible, because that package still uses 1.37.0 :(

For this use case, I made my own flake which only provides the driver and playwright-test. It has no instruction on how to use it currently, but it auto updates nightly and I use it in a big project. 

https://github.com/pietdevries94/playwright-web-flake

OliverJAsh commented:
That's great, thank you! I would love to see an example of how to use it. I've tried a few things but no luck so far.

pietdevries94 commented:
> That's great, thank you! I would love to see an example of how to use it. I've tried a few things but no luck so far.

I have created a new issue in the repo, so we don't polute this issue any further :)

https://github.com/pietdevries94/playwright-web-flake/issues/1

natsukium commented:
from ofborg
```
error: hash mismatch in fixed-output derivation '/nix/store/wd0narn4qcxhadam0f4hj7jwd42d78fk-source.drv':
         specified: sha256-AzH1rZFqEH8sovZZfJykvsEmCedEZWigQFHWHl6/PdE=
            got:    sha256-7egK76A3+C+JPbCNFXDd4qTjepBRSZgtQmFrE/jWJN4=
building '/nix/store/szihii9h9yz7qa6m984f1jcg6a726s75-playwright-driver-1.37.0.drv'...
```

natsukium commented:
I think the driver issue and this PR are independent of each other.
What do you think about merging this PR after the hash is fixed and continuing with #249738 for the driver?

Commit:
playwright: 1.34.3 > 1.37.0

pietdevries94 commented:
I think that's a good idea :)

natsukium commented:
Result of `nixpkgs-review pr 246551` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>30 packages marked as broken and skipped:</summary>
  <ul>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python310Packages.batchspawner</li>
    <li>python310Packages.batchspawner.dist</li>
    <li>python310Packages.dockerspawner</li>
    <li>python310Packages.dockerspawner.dist</li>
    <li>python310Packages.jupyterhub</li>
    <li>python310Packages.jupyterhub-ldapauthenticator</li>
    <li>python310Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python310Packages.jupyterhub-systemdspawner</li>
    <li>python310Packages.jupyterhub-systemdspawner.dist</li>
    <li>python310Packages.jupyterhub-tmpauthenticator</li>
    <li>python310Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python310Packages.jupyterhub.dist</li>
    <li>python310Packages.oauthenticator</li>
    <li>python310Packages.oauthenticator.dist</li>
    <li>python311Packages.batchspawner</li>
    <li>python311Packages.batchspawner.dist</li>
    <li>python311Packages.dockerspawner</li>
    <li>python311Packages.dockerspawner.dist</li>
    <li>python311Packages.jupyterhub</li>
    <li>python311Packages.jupyterhub-ldapauthenticator</li>
    <li>python311Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python311Packages.jupyterhub-systemdspawner</li>
    <li>python311Packages.jupyterhub-systemdspawner.dist</li>
    <li>python311Packages.jupyterhub-tmpauthenticator</li>
    <li>python311Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python311Packages.jupyterhub.dist</li>
    <li>python311Packages.oauthenticator</li>
    <li>python311Packages.oauthenticator.dist</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>changedetection-io</li>
    <li>changedetection-io.dist</li>
    <li>playwright (python310Packages.playwright)</li>
    <li>playwright-test</li>
    <li>playwright.dist (python310Packages.playwright.dist)</li>
    <li>python310Packages.pycookiecheat</li>
    <li>python310Packages.pycookiecheat.dist</li>
    <li>python310Packages.pytest-playwright</li>
    <li>python310Packages.pytest-playwright.dist</li>
    <li>python311Packages.playwright</li>
    <li>python311Packages.playwright.dist</li>
    <li>python311Packages.pycookiecheat</li>
    <li>python311Packages.pycookiecheat.dist</li>
    <li>python311Packages.pytest-playwright</li>
    <li>python311Packages.pytest-playwright.dist</li>
    <li>shot-scraper</li>
    <li>shot-scraper.dist</li>
  </ul>
</details>
